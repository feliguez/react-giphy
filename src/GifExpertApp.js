import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({ categoryList = [] }) => {
  const [categories, setCategories] = useState(categoryList);

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />
      <AddCategory setCategories={setCategories} />
      {categories.map((category) => (
        <GifGrid category={category} key={category} />
      ))}
    </>
  );
};

export default GifExpertApp;
