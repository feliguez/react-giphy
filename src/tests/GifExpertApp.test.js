import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Testing <GifExpertApp/>', () => {
  const wrapper = shallow(<GifExpertApp />);

  test('should match to snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should show a category list', () => {
    const categories = ['goku', 'marvel'];
    const wrapper = shallow(<GifExpertApp categoryList={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length)
  });
});
