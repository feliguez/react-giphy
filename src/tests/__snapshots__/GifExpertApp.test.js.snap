// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifExpertApp/> should match to snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
</Fragment>
`;

exports[`Testing <GifExpertApp/> should show a category list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <GifGrid
    category="goku"
    key="goku"
  />
  <GifGrid
    category="marvel"
    key="marvel"
  />
</Fragment>
`;
